astropy/io/votable/tree.py

```diff
@@ -3299,7 +3300,10 @@ def _write_binary(self, mode, w, **kwargs):
                     for i, converter in fields_basic:
                         try:
                             chunk = converter(array_row[i], array_mask[i])
-                            assert type(chunk) == bytes
```

Comment:
This is a code logic change from requiring `bytes` type exactly to requiring a `bytes` class or subclass. Probably OK but a votable maintainer needs to confirm.

Reply:
Thinking more, this PR should not change logic. So just replace this line with:
```
if type(chunk) != bytes:
    raise AssertionError
```